cadence/contracts/utils/EVMAbiHelpers.cdc

```diff
@@ -0,0 +1,143 @@
+import "EVM"
+
+access(all) contract EVMAbiHelpers {
+    // ===== Utilities: concat & padding =====
+    access(all) fun concat(_ parts: [[UInt8]]): [UInt8] {
+        var out: [UInt8] = []
+        for p in parts { out = out.concat(p) }
+        return out
+    }
+
+    access(all) fun leftPad32(_ b: [UInt8]): [UInt8] {
+        let need = 32 - b.length
+        if need <= 0 { return b.slice(from: b.length - 32, upTo: b.length) }
+        var out: [UInt8] = []
+        var i = 0
+        while i < need { out.append(0); i = i + 1 }
+        return out.concat(b)
+    }
+
+    access(all) fun rightPadTo32(_ b: [UInt8]): [UInt8] {
+        let rem = b.length % 32
+        if rem == 0 { return b }
+        let pad = 32 - rem
+        var out = b
+        var i = 0
+        while i < pad { out.append(0); i = i + 1 }
+        return out
+    }
+
+    // ===== UInt256 & words =====
+    access(all) fun beBytesN(_ v: UInt256, _ n: Int): [UInt8] {
+        var x = v
+        var out: [UInt8] = []
+        var i = 0
+        while i < n {
+            let byte = UInt8(x & 0xff)
+            out.insert(at: 0, byte)    // big-endian
+            x = x >> 8
+            i = i + 1
+        }
+        return out
+    }
+
+    access(all) fun abiWord(_ v: UInt256): [UInt8] { return self.leftPad32(self.beBytesN(v, 32)) }
+
+    // ===== Primitive encoders =====
```

Comment:
Since primitives are encodable via `EVM.encodeABI()`, can we just use the native encoding to prevent duplicate code & introducing additional risk surface area?